### PR TITLE
Fix OCR test flakes

### PR DESCRIPTION
**Pull request by connorwstein**

### CODE REVIEW

**Review by vyzaldysanchez — APPROVED** [review]

**Review by samsondav — APPROVED** [review]
Nice one

File: core/internal/features_test.go [hunk 1]
```diff
@@ -752,7 +752,7 @@ observationSource = """
 					2, 7, apps[i].JobORM(), time.Minute, time.Second)
 				jb, err := pr[0].Outputs.MarshalJSON()
 				require.NoError(t, err)
-				assert.Equal(t, []byte(fmt.Sprintf("[\"%d\"]", 10*i)), jb)
+				assert.Equal(t, []byte(fmt.Sprintf("[\"%d\"]", 10*i)), jb, "pr[0] %+v pr[1] %+v", pr[0], pr[1])
```
[the remark this answers is not in the document]
Comment (connorwstein) [reply core/internal/features_test.go:755]: this debug info?

File: core/internal/cltest/cltest.go [hunk 2]
```diff
@@ -979,7 +979,7 @@ func WaitForPipeline(t testing.TB, nodeID int, jobID int32, expectedPipelineRuns
 
 		var matched []pipeline.Run
 		for _, pr := range prs {
-			if !pr.State.Finished() && pr.State != state {
+			if !pr.State.Finished() || pr.State != state {
```
Comment (connorwstein) [inline core/internal/cltest/cltest.go:982]: we want to exclude the run if its either incomplete OR if its not at the target state

**Review by jmank88 — APPROVED** [review]

**Review by pinebit — APPROVED** [review]

**Review by prashantkumar1982 — APPROVED** [review]
Oops, this was my mistake :(
Thanks for catching!

**Pull request merged**